Pipeline/Preprocessing/YouTubeMetaExtraction.py

```diff
@@ -0,0 +1,185 @@
+import pytube
+import argparse
+import csv
+import re
+
+KNOWN_CHANNELS = ['Вечерний Ургант', 'НТВ']
+
+
+class YouTubeMetaExtraction:
+
+    def __init__(self, url):
+        self.__video = pytube.YouTube(url)
+        self.__url = url
+
+    def get_description(self):
+        """
+        Getter for the description of the video
+        :return: description (str)
+        """
+        return self.__video.description
+
+    def get_title(self):
+        """
+        Getter for the title of the video
+        :return: title (str)
+        """
+        return self.__video.title
+
+    def get_channel_name(self):
+        """
+        Getter for the author of the video
+        :return: author (str)
+        """
+        return self.__video.author
+
+    def get_html(self):
+        """
+        Getter for the html code of a page with a video
+        :return: html code (str)
+        """
+        return pytube.request.get(self.__url)
+
+    def get_time_codes_list_from_description(self):
+        """
+        Getter for a list with time codes and names from description
+        :return: list with time codes and names (list)
+        """
+        pattern = '[0-9]+:[0-9]+[^\n\r]+'
+        codes = re.findall(pattern, self.__video.description)
+        return codes
+
+    def get_songs_list_from_description(self):
+        """
+        Getter for a list with the names of parts from the description
+        :return: list with names (list)
+        """
+        songs = self.get_time_codes_list_from_description()
+        for i in range(len(songs)):
+            songs[i] = re.sub('[0-9:]+[\W]+', '', songs[i])
+        return songs
+
+    def get_caption(self, language_code):
+        """
+        Getter for the caption to the video
+        :param language_code: language code
+        :return: caption (str)
+        """
+        cap = self.__video.captions.get_by_language_code(language_code)
+        assert cap is not None, "Invalid language code"
```

Comment:
Are other errors possible?
From my point of view, if you make a specific assertion about an argument, you need to check it directly

Reply:
added more checks